go-controller/pkg/node/node_ip_handler_linux.go

```diff
@@ -196,11 +196,15 @@ func (c *addressManager) updateNodeAddressAnnotations() {
 	}
 
 	// get updated interface IP addresses for the gateway bridge
+	c.gatewayBridge.Lock()
```

Comment:
Move into a helper function and call `defer c.gatewayBridge.Lock()`? May be overkill, but my preference would be unlocking in a single location and ideally in a defer
~~~
ifAddrs, err := c.getNetworkInterfaceIPAddresses()
if err != nil {
    klog.Errorf("Failed to get gateway bridge IP addresses: %v", err)
         return
    }
}  
~~~

~~~
// getNetworkInterfaceIPAddresses sets and returns the bridge's current ifAddresses.
func (c *addressManager) getNetworkInterfaceIPAddresses() ([]*net.IPNet, error) {
  c.gatewayBridge.Lock()
  defer c.gatewayBridge.Unlock()
  ifAddrs, err := getNetworkInterfaceIPAddresses(c.gatewayBridge.bridgeName)
  if err == nil {
    c.gatewayBridge.ips = ifAddrs
  } 
 return ifAddrs, err
}
~~~

Reply:
@andreaskaris Thanks for the review! applied the suggestions.